### PR TITLE
Feature/edit history subgraph

### LINKED ISSUE #282: Asset edit history
- Use TheGraph to get history
- Display history in asset details

### PR DESCRIPTION
**Pull request by claudiaHash**
Fixes #282 .

Changes proposed in this PR:

- fetched Subgraph data for edit history component
- included EditHistory component in the asset content
- removed Published and Updated sections

### CODE REVIEW

**Comment by kremalicious** [comment]
woah, it just works! That's already shippable although it's in draft mode

Looking at https://market-git-feature-edit-history-subgraph-oceanprotocol.vercel.app/asset/did:op:fcB47f5781F14Ed7E032BD395113b84C897aA23f

Screen Shot 2021-03-04 at 17 01 20

File: src/components/organisms/AssetContent/EditHistory.module.css [hunk 1]
```diff
@@ -7,6 +7,7 @@
 .history {
   font-size: var(--font-size-small);
   padding-left: 1.25rem;
+  margin-bottom: calc(var(--spacer) );
```
Comment (kremalicious) [inline src/components/organisms/AssetContent/EditHistory.module.css:10]: it looks better without this bottom spacing I think. Then bottom & left white space of content box is more consistent

Comment (claudiaHash) [reply src/components/organisms/AssetContent/EditHistory.module.css:10]: Ok then, I'll remove the bottom spacing

**Comment by kremalicious** [comment]
as for the ordering, instead of figuring out to sort the `datatokens.updates` data in js, we might want to switch to [`metadataUpdates` query](https://subgraph.rinkeby.oceanprotocol.com/subgraphs/name/oceanprotocol/ocean-subgraph/graphql?query=%7B%0A%20%20metadataUpdates(where%3A%20%7BdatatokenId%3A%20%220x0006675fcb062eef3d9e9f71647151896391f002%22%7D)%20%7B%0A%20%20%20%20id%0A%20%20%20%20tx%0A%20%20%20%20timestamp%0A%20%20%20%20userAddress%0A%20%20%7D%0A%7D%0A) for this, since this we can sort in GraphQL then.

File: src/components/organisms/AssetContent/MetaFull.tsx [hunk 2]
```diff
@@ -23,10 +23,6 @@ export default function MetaFull(): ReactElement {
       /> */}
 
       {/* TODO: remove those 2 date items here when EditHistory component is ready */}
```
Comment (kremalicious) [inline src/components/organisms/AssetContent/MetaFull.tsx:25]: so looks like we can remove this code comment then

Comment (claudiaHash) [reply src/components/organisms/AssetContent/MetaFull.tsx:25]: done! think it's good to go now

Comment (kremalicious) [reply src/components/organisms/AssetContent/MetaFull.tsx:25]: hey, you found another comment too. I meant this comment here which we can also remove

File: src/components/organisms/AssetContent/EditHistory.tsx [hunk 3]
```diff
@@ -4,52 +4,56 @@ import { useAsset } from '../../../providers/Asset'
...
+      tx
+      updates(orderBy: timestamp, orderDirection: desc) {
+        id
+        tx
+        timestamp
+      }
+    }
+  }
+`
 interface Receipt {
```
Comment (kremalicious) [inline src/components/organisms/AssetContent/EditHistory.tsx:22]: we should remove this custom interface and rely on the generated Apollo typings.

Tested this import:

```js
import { ReceiptData_datatokens_updates as ReceiptData } from '../../../@types/apollo/ReceiptData'
```

Then use `ReceiptData` as the type in the code. And with that we already catch a minor issue which is the point of the typings:

Screen Shot 2021-03-04 at 17 53 37

Comment (claudiaHash) [reply src/components/organisms/AssetContent/EditHistory.tsx:22]: ok, on it

**Review by kremalicious — APPROVED** [review]
all perfect and with correct order in my testing, ship it!

**Pull request merged**